File: CS/OpenedWindows/DocumentsConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Data;
using System.Globalization;
using DevExpress.Xpf.Docking;
using System.Collections.ObjectModel;
using DevExpress.Xpf.Layout.Core;

namespace OpenedWindows
{
    public class DocumentsConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            DockLayoutManager manager = value as DockLayoutManager;
            ObservableCollection<object> documents = new ObservableCollection<object>();
            foreach (BaseLayoutItem item in manager.GetItems())
                if (item.ItemType == LayoutItemType.Document && !item.Closed)
                    documents.Add(item);
            return documents;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return value;
        }
    }
}

File: VB/OpenedWindows/DocumentsConverter.vb
Imports System
Imports System.Windows.Data
Imports System.Globalization
Imports DevExpress.Xpf.Docking
Imports System.Collections.ObjectModel
Imports DevExpress.Xpf.Layout.Core

Namespace OpenedWindows

    Public Class DocumentsConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim manager As DockLayoutManager = TryCast(value, DockLayoutManager)
            Dim documents As ObservableCollection(Of Object) = New ObservableCollection(Of Object)()
            For Each item As BaseLayoutItem In manager.GetItems()
                If item.ItemType = LayoutItemType.Document AndAlso Not item.Closed Then documents.Add(item)
            Next

            Return documents
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Return value
        End Function
    End Class
End Namespace
